src/Components/Users/UserProfile.tsx

```diff
@@ -0,0 +1,421 @@
+import React, { useState, useCallback, useReducer } from 'react';
+import PageTitle from '../Common/PageTitle'
+import { InputLabel, Button } from '@material-ui/core';
+import CheckCircleOutlineIcon from '@material-ui/icons/CheckCircleOutline';
+import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';
+import ExpansionPanel from '@material-ui/core/ExpansionPanel';
+import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
+import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
+import Typography from '@material-ui/core/Typography';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import { statusType, useAbortableEffect } from "../../Common/utils";
+import { GENDER_TYPES } from "../../Common/constants";
+import { useDispatch, useSelector } from "react-redux";
+import { getUserDetails, updateUserDetails } from "../../Redux/actions";
+import { PhoneNumberField, SelectField, TextInputField } from "../Common/HelperInputFields";
+import { parsePhoneNumberFromString } from 'libphonenumber-js';
+import { validateEmailAddress } from "../../Common/validation";
+import * as Notification from "../../Utils/Notifications.js";
+
+const useStyles = makeStyles((theme: Theme) =>
+    createStyles({
+        root: {
+            width: '100%',
+        },
+        heading: {
+            fontSize: theme.typography.pxToRem(15),
+            flexBasis: '33.33%',
+            flexShrink: 0,
+        },
+        secondaryHeading: {
+            fontSize: theme.typography.pxToRem(15),
+            color: theme.palette.text.secondary,
+        },
+    }),
+);
+const initForm: any = {
+    first_name: "",
+    last_name: "",
+    age: "",
+    gender: "",
+    email: "",
+    phone_number: "",
```

Comment:
Codacy found an issue: [Identifier 'phone_number' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5811785)